gammapy/estimators/flux_map.py

```diff
@@ -0,0 +1,399 @@
+            fp[norm_quantity] = res
+
+        # TODO: add support of norm and stat scan
+
+        rows = []
+        for idx, energy in enumerate(self.e_ref):
+            result = dict()
+            result["e_ref"] = energy
+            result["e_min"] = self.e_min[idx]
+            result["e_max"] = self.e_max[idx]
+            result["ref_dnde"] = ref[idx]
+            result["norm"] = fp["norm"][idx]
+            for quantity in self._available_quantities:
+                norm_quantity = f"norm_{quantity}"
+                result[norm_quantity] = fp[norm_quantity][idx]
+            rows.append(result)
+        table = table_from_row_data(rows=rows, meta={"SED_TYPE": "likelihood"})
+        return FluxPoints(table)
+
+    def to_dict(self, sed_type="likelihood"):
+        """Return maps in a given SED type in the form of a dictionary.
+
+        Parameters
+        ----------
+        sed_type : str
+            sed type to convert to. Default is `Likelihood`
+
+        Returns
+        -------
+        map_dict : dict
+            dictionary containing the requested maps.
+        """
+        if sed_type == "likelihood":
+            map_dict = self.data
+        else:
+            map_dict = {}
+            for entry in REQUIRED_MAPS[sed_type]:
+                map_dict[entry] = self.__getattribute__(entry)
+
+            for entry in OPTIONAL_MAPS[sed_type]:
+                try:
+                    map_dict[entry] = self.__getattribute__(entry)
+                except KeyError:
+                    pass
+
+            for key in self.data.keys() - (REQUIRED_MAPS["likelihood"] + OPTIONAL_MAPS["likelihood"]):
+                map_dict[key] = self.data[key]
+
+        return map_dict
+
+    def write(self, filename, filename_model=None, overwrite=False, sed_type="likelihood"):
+        """Write flux map to file.
+
+        Parameters
+        ----------
+        filename : str
+            Filename to write to.
+        filename_model : str
+            Filename of the model (yaml format).
+            If None, keep string before '.' and add '_model.yaml' suffix
+        overwrite : bool
+            Overwrite file if it exists.
+        sed_type : str
+            sed type to convert to. Default is `Lielihood`
+        """
+        filename = make_path(filename)
+
+        if filename_model is None:
+            name_string = filename.as_posix()
+            for suffix in filename.suffixes:
+                name_string.replace(suffix,'')
+            filename_model = name_string + '_model.yaml'
+        filename_model=make_path(filename_model)
+
+        hdulist = self.to_hdulist(sed_type)
+
+        models = Models(self.reference_model)
+        models.write(filename_model, overwrite=overwrite)
+        hdulist[0].header['MODEL'] = filename_model.as_posix()
+
+        hdulist.writeto(str(make_path(filename)), overwrite=overwrite)
+
+    def to_hdulist(self, sed_type="likelihood", hdu_bands=None):
```

Comment:
I would much prefer convention over configuration here and not allow to define the name of the `BANDS` hdu, once it's introduce people will use it and modify the names as they like. Let's be strict on write but forgiving on read...

I guess the option is to allow for a single band HDU only? Maybe we just offer this behaviour, I'm not sure whether we need the case with all individual band HDUs defined. I'll think about once again...